.ci/prepare-settings.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -e
+
+mkdir -p .ci-temp
+cp .ci/release-settings.xml .ci-temp/
+
+checkForVariable() {
+  VAR_NAME=$1
+  if [ -v "${!VAR_NAME}" ]; then
+    echo "Error: Define $1 environment variable"
+    exit 1
+  fi
+}
+
+checkForVariable "SONATYPE_USER"
+checkForVariable "SONATYPE_PWD"
+checkForVariable "GPG_PASSPHRASE"
+checkForVariable "GPG_KEY"
+
+replace() {
+  sed -i "s/$1/$2/g" .ci-temp/release-settings.xml
+}
+
+replace SONATYPE_USER
+replace SONATYPE_PWD
+replace GPG_PASSPHRASE
+replace GPG_KEY
```

Comment:
Shouldn't these names be more Checkstyle specific? IE: CS_GPG_KEY or CHECKSTYLE_GPG_KEY

Reply:
I am not sure for now to put CS as prefix for all.

Reply:
Aren't these just environment variables the user has set on their machine going by `checkForVariable`? Are these environment variables needed some place else besides for this script?

Reply:
Just a variables, not used in other scripts.

We never named variables with CS prefix, I not sure that we should start now.